### ISSUE TITLE
Global feedback on the current state of the doc

### ISSUE THREAD

**Issue by tmartin** [php-lug/documentation#9]
Yo! Here is my feedback on the doc.

First of all, great work! :+1: 

I know how hard it is to make oneself clear after writing this amount of code. The hardest part is to clarify the basic concepts. IMO, we should put our energy into this: clarifying the core knowledge. Let the users sit down to read about the basics. Then we can throw code at their faces. :)

This can be done by adding more in the introduction of each component or bundle, before going to the usage and the features. I would suggest taking a look at the [Expression Language component doc](http://symfony.com/doc/current/components/expression_language/introduction.html) as it is a good example of what I mean (not too long, not too short).

Every section of the doc suffers from this lack of stepping back. My advise on this: you should try to sell it to me by explaining what we can do with it and emphase on what's new and amazing. After that we can try to put it down on paper. What do you think? :)

Below are some more specific comments, but I think the global message is more important. I've also made a small PR about some typos I've found: #8

## Global

**1.** In the index file, I think we should orient the user by adding a short description of each section (eg. "the book", "the cookbook", etc.). This way, he knows quickly where to start and where to find each kind of information (purely technical, or more guided walkthrough like a tuto).

**2.** Can we make the "installation" section of each component more concise ? In the meantime, we could add a line about installation directly from Gihub. Like the ones in the Symfony doc (it's a copy-paste for every component) :

> You can install the component in 2 different ways:
> * Install it via Composer (symfony/console on Packagist);
> * Use the official Git repository (https://github.com/symfony/console).
> 
> Then, require the `vendor/autoload.php` file to enable the autoloading mechanism provided by Composer. Otherwise, your application won't be able to find the classes of this Symfony component.

## Registry component

**3.** In the "usage" section, I think the first paragraph should be rewritten.

> The registry component is designed around the ServiceRegistryInterface which is itself designed around the ArrayAccess, Countable and IteratorAggregate interfaces. This provides you a very consistent API to manipulate your services.

The first part of the first sentence adds complexity by introducing an interface that you will never speak of again in the doc. The second half repeats what you have said in your introduction. The second sentence adds a new concept that has not been described in the intro. You were talking about abstraction and now about an API for services. I must say that as a new user I'm a bit confused. :)

**4.** In the "Create your registry" section, could you rephrase the note about the second argument? I don't understand the part about throwing exceptions.

**5.** IMO the section of code examples could be made more clear if you directly stated that it behaves like an array, because it implements ArrayAccess. Knowing that, the user can understand how straightforward your examples are.

**6.** In the end, I think I miss the point of this component. Like I said in my "global" comment, I think it's due to the lack of details about the basic concepts. I'm left with questions like:
* What is a service in this context?
* Are we in symfony or an independant component?
* I don't know where to use this. Can you show me some use cases?

## Storage component

**7.** Same comment as before. We should add a more detailed introduction. Talking about the design is great, but only *advanced code warriors +9* will understand the advantages of such a design just by reading this sentence. :)

**8.** It's even more obvious in this component that we are missing the key arguments that makes it an awesome piece of code. We should take the time to explain why using this method or this method will help the developer in his daily life.

## Bundles

I've read each bundle doc and the same issues arose.

**Comment by egeloen** [comment]
Now I have mostly finished bootstraping tools around the project, I come back here. Let's first focus on the Registry component/bundle.
